# Actividad 1: Tablero de ajedrez y granos de arroz

In [167]:
import plotly.graph_objects as go
from plotly.subplots import make_subplots

from numpy import *
import pandas as pd
pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)

casilla = arange(1,65,1, dtype='object')
arroz = power(2,casilla)
casilla_arroz = pd.DataFrame(arroz,columns = ['Numero de granos de arroz'], index = casilla)
casilla_arroz.index.name = 'Numero de casilla'

display(casilla_arroz)

Numero de granos de arroz
Numero de casilla                          
1                                         2
2                                         4
3                                         8
4                                        16
5                                        32
6                                        64
7                                       128
8                                       256
9                                       512
10                                     1024
11                                     2048
12                                     4096
13                                     8192
14                                    16384
15                                    32768
16                                    65536
17                                   131072
18                                   262144
19                                   524288
20                                  1048576
21                                  2097152
22                                  4194304
23                                  8388608
24                                 16777216
25                                 33554432
26                                 67108864
27                                134217728
28                                268435456
29                                536870912
30                               1073741824
31                               2147483648
32                               4294967296
33                               8589934592
34                              17179869184
35                              34359738368
36                              68719476736
37                             137438953472
38                             274877906944
39                             549755813888
40                            1099511627776
41                            2199023255552
42                            4398046511104
43                            8796093022208
44                           17592186044416
45                           35184372088832
46                           70368744177664
47                          140737488355328
48                          281474976710656
49                          562949953421312
50                         1125899906842624
51                         2251799813685248
52                         4503599627370496
53                         9007199254740992
54                        18014398509481984
55                        36028797018963968
56                        72057594037927936
57                       144115188075855872
58                       288230376151711744
59                       576460752303423488
60                      1152921504606846976
61                      2305843009213693952
62                      4611686018427387904
63                      9223372036854775808
64                     18446744073709551616

A la casilla 25 le corresponden 33554432 granos, mientras que a la 64 le corresponen 18446744073709551616 granos.

Como el numero de granos de cada casilla es el doble al de la casilla anterior a la casilla $n$ le corresponden $2^n$ granos. 

# Actividad 2

In [233]:
dia = arange(1,31,1, dtype='object')
A_dia = []
B_dia = []
A_acumulado = []
B_acumulado = []
total_A = 0
total_B = 0
for i in dia:
    B_dia.append(10)
    A_dia.append(0.1*2**(i-1))
    total_B+=10
    total_A+=0.1*2**(i-1)
    B_acumulado.append(total_B)
    A_acumulado.append(total_A)


df = pd.DataFrame((zip(A_dia,A_acumulado,B_dia,B_acumulado)), index = dia)
df.columns = ['Obtenido por A en el dia',
              'Obtenido por A acumulado',
              'Obtenido por B en el dia',
              'Obtenido por B en el dia']
display(df)




Obtenido por A en el dia  Obtenido por A acumulado  Obtenido por B en el dia  Obtenido por B en el dia
1                        0.1                       0.1                        10                        10
2                        0.2                       0.3                        10                        20
3                        0.4                       0.7                        10                        30
4                        0.8                       1.5                        10                        40
5                        1.6                       3.1                        10                        50
6                        3.2                       6.3                        10                        60
7                        6.4                      12.7                        10                        70
8                       12.8                      25.5                        10                        80
9                       25.6                      51.1                        10                        90
10                      51.2                     102.3                        10                       100
11                     102.4                     204.7                        10                       110
12                     204.8                     409.5                        10                       120
13                     409.6                     819.1                        10                       130
14                     819.2                    1638.3                        10                       140
15                    1638.4                    3276.7                        10                       150
16                    3276.8                    6553.5                        10                       160
17                    6553.6                   13107.1                        10                       170
18                   13107.2                   26214.3                        10                       180
19                   26214.4                   52428.7                        10                       190
20                   52428.8                  104857.5                        10                       200
21                  104857.6                  209715.1                        10                       210
22                  209715.2                  419430.3                        10                       220
23                  419430.4                  838860.7                        10                       230
24                  838860.8                 1677721.5                        10                       240
25                 1677721.6                 3355443.1                        10                       250
26                 3355443.2                 6710886.3                        10                       260
27                 6710886.4                13421772.7                        10                       270
28                13421772.8                26843545.5                        10                       280
29                26843545.6                53687091.1                        10                       290
30                53687091.2               107374182.3                        10                       300

$a)$ Claramente la situacion benficio a A

$b)$ En el septimo dia B le entrego a A 25 pesos mientras que A le entrego a B 70 pesos

$c,d,e)$ En el dia 15, B tiene 150 pesos, mientras que A tiene 6553.40 pesos
En el dia 30, B tiene 300 pesos, mientras que A tiene  214748364.60 pesos,
esto se debe a que el comportamiento de la funcion de lo que obtiene B es lineal, mientras que
el comportamiento de lo que obtiene A es exponencial.

$f)$ El dinero entregado en el dia $n$ es:

Entregado por A: $ a_n = 10 * n $

Entregado por B: $ b_n = 0.1 * 2^n$

# Grafico

In [220]:
fig = go.Figure()
fig.add_trace(go.Scatter(name = "Obtenido por A cada dia", x = dia, y = A_dia))
fig.add_trace(go.Scatter(name = "Obtenido por B cada dia", x = dia, y = B_dia))

fig.show()

# Actividad del horno

In [223]:
data_horno = [14,23,18,15,14,16,15,20,16,17,20,17,19,15,17,18,22,16,
17,15,17,20,16,15,17,16,17,17,18,15,19,21,17,16,17,16,
19,17,21,18,20,17,17,16,22,17,15,17,16,19,18]

print('Se rompieron a los 14 minutos: ', data_horno.count(14))
print('Se rompieron a los 19 minutos: ',data_horno.count(19))
print('Se rompieron a los 15 minutos: ',data_horno.count(14))



Se rompieron a los 14 minutos:  2
Se rompieron a los 19 minutos:  4
Se rompieron a los 15 minutos:  2


# Actividad 4: 

In [44]:
X = arange(-4,5,1)
Y = power(1/2,X)

df = pd.DataFrame(zip(X,Y),columns = ['X','Y'])
df

X        Y
0 -4  16.0000
1 -3   8.0000
2 -2   4.0000
3 -1   2.0000
4  0   1.0000
5  1   0.5000
6  2   0.2500
7  3   0.1250
8  4   0.0625

In [222]:
fig = go.Figure()
fig.add_trace(go.Scatter(name = "Funcion Y(x) = (1/2)^x", x = X, y = Y))
fig.show()

$a)$ Es una curva que tiende a cero a medida que $X$ crece y diverge a medida de que $X$ decrece.

$b)$ Corta al eje $Y$ en $X$ = 0. Ese punto se llama ordenada al origen.

$c)$ La curva es siempre positiva ya que $\left(\frac{1}{2}\right)^x > 0$ por lo que no corta al eje $X$ en ningun punto.

$d)$ El codominio tiene a infinito a medida de que el dominio tiene a infinito negativo

$e)$ El grafico decrece porque mientras mas grande es $X$, menor es $Y$

